The CDISC safety analysis identified a dose-dependent adverse event
signal with a hazard ratio of up to 5.03 (highest dose vs. placebo),
based on 254 subjects across three arms. As a pilot study, this
analysis was not specifically powered around the comparison in
question.

This notebook illustrates how that finding could inform the design of
a dedicated confirmatory study. The hazard ratio (5.03) is the
observed effect from the prior analysis. The median time-to-event and
study timeline used below are illustrative assumptions for a new,
hypothetical study, since a confirmatory trial would follow its own
design rather than replicate the pilot.

In [1]:
from power_calc.survival import SurvivalEndpoint

trial = SurvivalEndpoint(accrual_period=12.0, follow_up_period=24.0)

In [4]:
def print_survival_result(result):
    print(f"Events needed:                 {result['events_needed']:.1f}")
    print(f"Event probability (control):   {result['p_control']:.1%}")
    print(f"Event probability (treatment): {result['p_treat']:.1%}")
    print(f"Sample size per arm:           {result['n_control']:.1f}")
    print(f"Total sample size:             {result['n_total']:.1f}")

## Scenario 1: Sample Size Based on the Observed Effect (HR = 5.03)

Assumptions: a 24-month median time-to-event in the placebo arm, a
12-month accrual period, and 24 months of follow-up after the last
subject is enrolled.

In [6]:
result_observed = trial.closed_form_sample_size(
    hazard_ratio=5.03, control_median_survival=24.0, power=0.8,
)
print_survival_result(result_observed)

Events needed:                 12.0
Event probability (control):   57.7%
Event probability (treatment): 98.6%
Sample size per arm:           7.7
Total sample size:             15.4


The result is approximately 15 total subjects, substantially fewer
than the pilot's 254. This follows directly from the magnitude of the
effect: a hazard ratio of 5.03 represents a large difference between
arms, and large effects require comparatively few events to detect at
conventional power levels.

This result illustrates a broader point: required sample size is
highly sensitive to the assumed effect size. Confirmatory studies are
rarely powered around the most extreme estimate from a small pilot; a
more conservative assumption is typically used to account for the
possibility that the pilot estimate will not fully replicate.

## Scenario 2: Sample Size Under a Conservative Effect Assumption (HR = 1.5)

Same study timeline as above, with a more conservative assumed hazard
ratio of 1.5 rather than 5.03. This reflects a more defensible
planning assumption for a confirmatory trial.

In [7]:
result_conservative = trial.closed_form_sample_size(
    hazard_ratio=1.5, control_median_survival=24.0, power=0.8,
)
print_survival_result(result_conservative)

Events needed:                 191.0
Event probability (control):   57.7%
Event probability (treatment): 72.4%
Sample size per arm:           146.7
Total sample size:             293.4


This scenario requires approximately 293 total subjects, nearly twenty
times the sample size required under the observed effect. This
comparison illustrates the central point of this notebook: required
sample size for a confirmatory trial depends substantially on the
level of confidence placed in a pilot study's effect size estimate.
Evaluating that trade-off is a judgment this tool is intended to
support rather than replace.